### PR TITLE
fix passing user agent if doesn't exist

### PR DESCRIPTION
**Pull request by evq**

### CODE REVIEW

**Comment by evq** [comment]
hmm. it seems better to me to pass the correct type in the first place versus cleaning it up after

**Comment by mrose17** [comment]
it seems better to me to fix it once in one place and as generally as possible.

**Comment by evq** [comment]
I agree, we should switch to a statically typed language :)

**Comment by mrose17** [comment]
as long as it is also a functional programming language, why not? (-;

but that train left the station two years ago...

**Comment by evq** [comment]
in liu of fixing this in the most general way, my thoughts here are that I would much prefer an error when calling something with the wrong type versus having the called code try to fix my types. having the code fixup the types by deleting keys is rather unintuitive.

**Comment by evq** [comment]
let me know if you disagree and still pref deleting keys, I'll change it if so

**Comment by mrose17** [comment]
i disagree with your approach. i greatly prefer the approach i outlined...

File: bat-utils/lib/extras-hapi.js [hunk 1]
```diff
@@ -156,6 +156,10 @@ const WreckProxy = (server, opts) => {
   if (!opts.headers) opts.headers = {}
   if (!opts.headers['user-agent'] && !opts.headers['User-Agent']) opts.headers['user-agent'] = wreckUA
 
+  underscore.keys(opts.headers).forEach((header) => {
+    if (typeof opt.headers[header] !== 'string') delete opt.headers[header]
```
Comment (evq) [inline bat-utils/lib/extras-hapi.js]: typo - will repush

**Pull request merged**